lib/cloudware.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+#==============================================================================
+# Copyright (C) 2017 Stephen F. Norledge and Alces Software Ltd.
+#
+# This file/package is part of Alces Cloudware.
+#
+# Alces Cloudware is free software: you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation, either version 3 of
+# the License, or (at your option) any later version.
+#
+# Alces Cloudware is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this package.  If not, see <http://www.gnu.org/licenses/>.
+#
+# For more information on the Alces Cloudware, please visit:
+# https://github.com/alces-software/cloudware
+#==============================================================================
+
+lib_dir = File.dirname(__FILE__)
+$LOAD_PATH << File.join(lib_dir, 'cloudware')
+ENV['BUNDLE_GEMFILE'] ||= File.join(lib_dir, '..', 'Gemfile')
+
+Thread.report_on_exception = false
```

Comment:
The default behaviour changed in ruby 2.5 to `true`:
https://ruby-doc.org/core-2.5.0/Thread.html#method-c-report_on_exception

Previously threads would raise the error when `thread.join` is called. This allowed the main thread to handle the error nicely and hides the trace back.

But with the change in ruby `2.5`, the error gets reported to the console if it isn't handled within the slave thread. This causes the traceback to get dumped to the console before the main thread can handle it.

Whilst it would be possible to handle the traceback within all our slave threads, it is a break in the current design pattern. This is why `Thread.report_on_exception` is being set to `false` which was the previous default in ruby `2.4`